### PR TITLE
Retarget STDIO to USART

### PR DESCRIPTION
**Pull request by JonatanAntoni**
- USART output can be retrieved via a Telnet connection.
- For CI USART output is sent to STDOUT as before.

### CODE REVIEW

File: basic/retarget_stdio.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ 
+  \param[in]   ch  Character to output
+  \return          The character written, or -1 on write error.
+*/
+int stdout_putchar (int ch) {
+  int32_t ret;
+
+  if (Driver_USART0.Send(&ch, 1U) == ARM_DRIVER_OK) {
+#ifdef __UVISION_VERSION
+    // add carriage return after each line feed
```
Comment (vovamarch) [inline basic/retarget_stdio.c:42]: Suggest to add clarification comment:
// workaround for fixing output formatting in default Windows Telnet client. Applied in uVision environment only.

**Review by vovamarch — COMMENTED** [review]
See some additional recommended comments.

File: basic/RTE/Device/SSE-300-MPS3/startup_fvp_sse300_mps3.c [hunk 2]
```diff
@@ -47,6 +50,7 @@ void Reset_Handler  (void) __NO_RETURN;
 #define DEFAULT_IRQ_HANDLER(handler_name)  \
 void __WEAK __NO_RETURN handler_name(void); \
 void handler_name(void) { \
+		exit(-1); \
```
[the remark this answers is not in the document]
Comment (JonatanAntoni) [reply basic/RTE/Device/SSE-300-MPS3/startup_fvp_sse300_mps3.c:53]: Hard to add a comment in place. I could add one above the `#define`. Would that help?

File: basic/retarget_stdio.c (hunk 1 above)
Comment (JonatanAntoni) [reply basic/retarget_stdio.c]: Done.

File: basic/RTE/Device/SSE-300-MPS3/startup_fvp_sse300_mps3.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (vovamarch) [reply basic/RTE/Device/SSE-300-MPS3/startup_fvp_sse300_mps3.c:53]: Sure, this would be also good.

**Review by vovamarch — APPROVED** [review]

**Pull request merged**